src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -3002,26 +3002,20 @@ void MacroAssembler::test_klass_is_empty_inline_type(Register klass, Register te
 
 void MacroAssembler::test_field_is_null_free_inline_type(Register flags, Register temp_reg, Label& is_null_free_inline_type) {
   movl(temp_reg, flags);
-  shrl(temp_reg, ResolvedFieldEntry::is_null_free_inline_type_shift);
-  andl(temp_reg, 0x1);
-  testl(temp_reg, temp_reg);
-  jcc(Assembler::notZero, is_null_free_inline_type);
+  btl(temp_reg, ResolvedFieldEntry::is_null_free_inline_type_shift);
+  jcc(Assembler::carrySet, is_null_free_inline_type);
```

Comment:
You also do not need `temp_reg`. Also, I think `testl(flags, 1 << ResolvedFieldEntry::is_null_free_inline_type_shift)` would be better than `btl` since `test` can macro-fuse with `jcc`.